### PR TITLE
feat(web): expose Web API compatible version of `next-auth`

### PR DESCRIPTION
**Pull request by balazsorban44**
Introducing `next-auth/web`, a fully Web API compatible version of NextAuth.js.

Supersedes #4299

This PR is based on #4769, which has to be merged first. (Diff is therefore currently bigger than it is supposed to be.)

- [x] `crypto` interop
- [x] `authorization` endpoint
- [x] `callback` (`token` & `userinfo` endpoints)
- [ ] Node.js/Web interop in core (this should probably not be needed. :thinking:)

### CODE REVIEW

File: packages/next-auth/package.json [hunk 1]
```diff
@@ -78,12 +80,16 @@
     "uuid": "^8.3.2"
   },
   "peerDependencies": {
+    "@panva/oauth4webapi": "1.2.0",
```
Comment (panva) [inline packages/next-auth/package.json:83]: From yesterday onward it's just `oauth4webapi`, `^1.2.1`.

File: packages/next-auth/src/core/lib/assert.ts [hunk 2]
```diff
@@ -94,6 +99,20 @@ export function assertConfig(params: {
   let hasTwitterOAuth2
 
   for (const provider of options.providers) {
+    if (provider.type === "oauth" && !provider.issuer) {
```
Comment (hmnd) [inline packages/next-auth/src/core/lib/assert.ts:102]:
```suggestion
    if (provider.type === "oauth" && (!provider.wellKnown || !provider.issuer)) {
```

`wellKnown` should provide the endpoint configs. This check currently breaks providers such as Azure AD who rely on `wellKnown`.

**Comment by cmoog** [comment]
This is great!

**Comment by aequasi** [comment]
Anything the community can help with testing on this?

**Comment by balazsorban44** [comment]
> Anything the community can help with testing on this?

@aequasi, I appreciate the interest, soon...! :pray: :wink:

**Review by ThangHuuVu — APPROVED** [review]
LFG 🚢

**Comment by aequasi** [comment]
Any docs on using this?

**Pull request merged**